### ISSUE TITLE
Update Graph view to display cut down Console view.

### ISSUE THREAD

**Issue by timbrown5** [jenkinsci/pipeline-graph-view-plugin#19]
Currently, the Graph View prints a dialogue box when a node is selected. It would be nice to update this to show a cutdown version of the console view (e.g. only Step logs applicable to that node).

**Comment by timja** [comment]
possibly or even just redirect to the console page with the correct node expanded

**Comment by amuniz** [comment]
I think Blue Ocean UX on this is quite good and we could try to replicate it here.
The current approach does scale very well:

![screenshot-1](https://user-images.githubusercontent.com/1017585/132840722-f2d72b39-869f-4b45-b302-81690b0b9065.png)

Showing the logs right below the stage view when a specific stage/parallel branch is selected solves this issue. I'll try to implement something if you think it is worth it.

**Comment by timja** [comment]
Interested in anything atm so go for it :)

**Comment by reddwarf69** [comment]
> I think Blue Ocean UX on this is quite good and we could try to replicate it here.

I have seen different, more or less related, issues. But I agree, replicating Blue Ocean is probably a good default course of action.
When using this pipeline:
```
pipeline {
  agent any

  stages {
    stage('Test') {
      steps {
        sh 'sleep 2'
        sh 'for I in $(seq 10000); do echo AAAAAAAAAA; done'
      }
    }
  }
}
```
With Blue Ocean I get:
![Blue Ocean](https://user-images.githubusercontent.com/1608276/189628766-9ccb5bf0-ce1b-4425-aa0c-21763ac2c379.png)

While with pipeline-graph-view I get
![Graph](https://user-images.githubusercontent.com/1608276/189628890-0134f7f4-cfd5-4b48-9fc9-c01b04b53aa4.png)
and
![Truncated](https://user-images.githubusercontent.com/1608276/189628914-ccbe978b-8bbb-4a71-ac42-f68c8b58f20b.png)

A few things, with Blue Ocean
- I have a lot of information in a single view, including for example the time each step took
- I have access to the complete log, even if it's over 150 KiB.
- The log of a single step is updated in real time. So I can, for example, focus on a specific parallel branch while the job is still building.
- I can use the graph to move between stages easily. With pipeline-graph-view the second I click on a stage, I'm moved to `/pipeline-console` with no easy access back to `/pipeline-graph`.

**Comment by timja** [comment]
> I have access to the complete log, even if it's over 150 KiB.

I think that's mostly this:
https://github.com/jenkinsci/pipeline-graph-view-plugin/issues/20

-----

Thanks for the feedback :)

**Comment by timbrown5** [comment]
My reasoning for not following the BlueOcean console view is that we wanted something that could make navigating a pipeline easier - hence the use of a tree view.
The inspiration was trying to replicate the functionality of the StepView in a way that allows for expandable sections from https://ci.jenkins.io/job/Plugins/job/pipeline-graph-view-plugin/job/main/56/flowGraphTable/

That said, I agree the console view is not in a nice state (it's more of a PoC). I'm also happy to be overruled if my use case is not common.

**Comment by timja** [comment]
not sure if you've checked the last couple of releases @timbrown5 but it's looking nicer now.

**Comment by timbrown5** [comment]
Sorry, I thought I replied to this.

I noticed it looks a lot better now (I think we are using an old version in our Jenkins). I like the background in newer Jenkins versions too 👍 
![Screenshot 2022-09-16 at 10 00 16](https://user-images.githubusercontent.com/4447764/190606061-8119d25b-9d23-475d-a94e-a0146db87df4.png)

Also not sure who added the pipeline-console.css, but it makes controlling the look much easier than I remember it being.
